### PR TITLE
fix ProteinQuantifier

### PR DESCRIPTION
**Pull request by timosachsenberg**
## Description

Make quantification not rely on same order of entries in consensusmap column headers and the experimental design file.

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

File: src/openms/source/ANALYSIS/QUANTITATION/PeptideAndProteinQuant.cpp [hunk 1]
```diff
@@ -683,20 +694,25 @@ namespace OpenMS
...
-        size_t sample = ed.getMSFileSection()[row].sample;
-        quantifyFeature_(f, fraction, sample, hit); // updates "stats_.quant_features"
+        const auto& h = consensus.getColumnHeaders().at(row);
+        const String c_fn = FileHandler::stripExtension(File::basename(h.filename)); // filename according to experimental design in consensus map
+        const size_t c_lab = h.getLabelAsUInt(consensus.getExperimentType());
+
+        // find entry in experimental design (ignore extension and folder) that corresponds to current column header entry
+        if (auto it = fileAndLabel2MSFileSectionEntry.find(c_fn + String(c_lab)); it != fileAndLabel2MSFileSectionEntry.end())
+        {
+          const String& e_fn = it->second.path;
```
Comment (timosachsenberg) [inline src/openms/source/ANALYSIS/QUANTITATION/PeptideAndProteinQuant.cpp]:
```suggestion
```

**Comment by timosachsenberg** [comment]
merge as a hot-fix

**Comment by timosachsenberg** [comment]
would probably make sense to move this helper to ExperimentalDesign

File: src/openms/source/ANALYSIS/QUANTITATION/PeptideAndProteinQuant.cpp [hunk 2]
```diff
@@ -683,20 +694,24 @@ namespace OpenMS
       PeptideHit hit = getAnnotation_(c.getPeptideIdentifications());
       for (auto const & f : c.getFeatures())
       {
-        // indices in experimental design are 0-based as the map indices
         //TODO MULTIPLEXED: needs to be adapted for multiplexed experiments
```
Comment (jpfeuffer) [inline src/openms/source/ANALYSIS/QUANTITATION/PeptideAndProteinQuant.cpp:697]: What about this Todo?

Comment (timosachsenberg) [reply src/openms/source/ANALYSIS/QUANTITATION/PeptideAndProteinQuant.cpp:697]: Good question. Looks correct to me now. quantifyFeature_ now uses the correct fraction group, fraction, and sample.

```
pep_quant_[seq].abundances[fraction][hit.getCharge()][sample] += feature.getIntensity();
```

We don't consider channel specific effects here and just sum independent of the used channel.

**Comment by timosachsenberg** [comment]
BTW now both feature intensity and spectral counting show the same pattern.

```
spectral counting:
PRT	P00761|TRYP_PIG	Trypsin - Sus scrofa (Pig).	...	2.0	0.0	0.0	
feature intensity:
PRT	P00761|TRYP_PIG	Trypsin - Sus scrofa (Pig).	...	2.0622054e07	0.0	0.0
```

Spectral counting did the mapping to the experimental design already based on the ms filename and was already correct.

**Pull request merged**